draft-api/src/main/scala/no/ndla/draftapi/service/WriteService.scala

```diff
@@ -866,5 +868,67 @@ trait WriteService {
           )
       }
     }
+
+    def copyRevisionDates(publicId: String): Try[Boolean] = {
+      taxonomyApiClient.getNode(publicId) match {
+        case Failure(_)     => Failure(NotFoundException(s"No topics with id ${publicId}"))
+        case Success(topic) =>
+          // These revisions should be put on all children
+          val revisionMeta = topic.contentUri match {
+            case Some(contentUri) =>
+              parseArticleIdAndRevision(contentUri) match {
+                case (Success(articleId), _) =>
+                  draftRepository.withId(articleId) match {
+                    case Some(article) => article.revisionMeta
+                    case _             => Seq.empty
+                  }
+                case _ => Seq.empty
+              }
+            case _ => Seq.empty
+          }
+          if (revisionMeta.nonEmpty) {
+            val children  = taxonomyApiClient.getChildNodes(publicId)
+            val resources = taxonomyApiClient.getChildResources(publicId)
+            children match {
+              case Success(topics) => topics.foreach(setRevisions(_, revisionMeta))
+              case _               =>
+            }
+            resources match {
+              case Success(resources) => resources.foreach(setRevisions(_, revisionMeta))
+              case _                  =>
+            }
+          }
+          Success(true)
+      }
+    }
+
+    def setRevisions(entity: Taxonomy[_], revisions: Seq[domain.RevisionMeta]): Unit = {
+      entity.contentUri match {
+        case Some(contentUri) =>
+          parseArticleIdAndRevision(contentUri) match {
+            case (Success(articleId), _) => updateArticleWithRevisions(articleId, revisions)
+            case _                       =>
+          }
+        case _ =>
+      }
+      if (classOf[Topic].isInstance(entity)) {
+        val resources = taxonomyApiClient.getChildResources(entity.id)
+        resources match {
+          case Success(resources) => resources.foreach(setRevisions(_, revisions))
+          case _                  =>
+        }
+      }
+    }
+
+    def updateArticleWithRevisions(articleId: Long, revisions: Seq[domain.RevisionMeta]): Unit = {
+      val existing = draftRepository.withId(articleId)
+      val updated = existing match {
+        case Some(article) =>
+          val revisionMeta = article.revisionMeta ++ revisions
+          Some(article.copy(revisionMeta = revisionMeta.distinct))
+        case None => None
+      }
+      updated.map(draftRepository.updateArticle(_))
+    }
```

Comment:
Siden denne i teorien kan feile så stemmer jeg for at vi returnerer en `Try`
```suggestion
    def updateArticleWithRevisions(articleId: Long, revisions: Seq[domain.RevisionMeta]): Try[_] = {
      draftRepository
        .withId(articleId)
        .traverse(article => {
          val revisionMeta = article.revisionMeta ++ revisions
          val toUpdate     = article.copy(revisionMeta = revisionMeta.distinct)
          draftRepository.updateArticle(toUpdate)
        })
    }
```